Library/Homebrew/formula_specialties.rb

```diff
@@ -20,15 +20,26 @@ def self.url(val)
 class AmazonWebServicesFormula < Formula
   # Use this method to peform a standard install for Java-based tools,
   # keeping the .jars out of HOMEBREW_PREFIX/lib
-  def install
+  def standard_install options={}
+    java_version = options[:java]
+    env = options[:env] || {}
+    env.merge! Language::Java.java_home_env(java_version)
     rm Dir['bin/*.cmd'] # Remove Windows versions
     libexec.install Dir['*']
-    bin.install_symlink Dir["#{libexec}/bin/*"] - ["#{libexec}/bin/service"]
+    Pathname.glob("#{libexec}/bin/*") do |file|
+      next if file.directory?
+      basename = file.basename
+      next if basename.to_s == "service"
+      (bin/basename).write_env_script file, env
+    end
+  end
```

Comment:
If we decide to kill `AmazonWebServicesFormula`, I would suggest to kill `Library/Homebrew/formula_specialties.rb` entirely.